lib/Echidna/Transaction.hs

```diff
@@ -250,6 +250,13 @@ shrinkTx tx'@(Tx c _ _ _ gp (C _ v) (C _ t, C _ b)) = let
     ]
   in join (uniform possibilities) <*> pure tx'
 
+mutateTx :: (MonadRandom m, Has GenDict x, MonadState x m, MonadThrow m) => Tx -> m Tx
+mutateTx (Tx (SolCall c) a b d w x y) = do f <- weighted [(mutate, 10), (skip, 90)] --FIXME: make a parameter for this?
+                                           f c
+                                         where mutate  z = mutateAbiCall z >>= \c' -> return $ Tx (SolCall c') a b d w x y
+                                               skip    _ = return $ Tx (SolCall c) a b d w x y
```

Comment:
this could be a bit error prone, you can bind `mutateTx tx@(Tx (SolCall c) _ _ _ _ _ _)` and do `skip    _ = pure tx` here

Reply:
We can't use `tx` because it shadows something from `EVM`. I will be using `t`.